Update docs (Closes #110)

hmgaudecker opened the pull request.

timmens commented:
I updated the .pre-commit-config.yaml to use modern formatters on the documentation. Unfortunately, this also changed a few non-documentation files.

timmens commented:
I updated the build process for the documentation. Now you can run ``make html`` and it executes the pytask script that generates the documentation relevant figures before running ``sphinx-build``. This is just an added line to the Makefile and can be easily reverted.

hmgaudecker commented:
> I updated the build process for the documentation. Now you can run `make html` and it executes the pytask script that generates the documentation relevant figures before running `sphinx-build`. This is just an added line to the Makefile and can be easily reverted.

Cool, thanks! I am not sure I fully understand why the figures are still present in the repo then? If they are re-created each time `make html` runs (as they should in an ideal world), we could just delete them?

timmens commented:
Yes, I missed that. They are deleted now!

hmgaudecker commented:
Hi @timmens, thanks again for all your work! I went through everything now, looks great! Just a couple of wishes / updates:
- In 51c12b I changed the LaTeX pipeline to what you would do in a serious project. This likely broke CI again (ignore the commit before that, had a stupid mistake in that one). We need to infer LaTeX deps automatically etc., can't tell users that they should change that on their own. I think the way forward would be to add some conditionality depending on whether we are in a CI environment (maybe we need to add a cookiecutter option that we tell users to ignore), that should also help with the inability to run the full test pipeline on local machines?
- Also in that commit, I removed the paths set for LaTeX (too much action at a distance... Better be explicit with paths in documents). However, the tables don't show up in my DAG anymore, not sure why, could you investigate, please?
- I think we should add the data to the project rather than downloading it. Else people will think pytask would check for changes in the remote data, too, which is not what is happening. Please also add the precise source to the "abstract" in `running_example`.

Thanks!!!!

hmgaudecker commented:
Yes, cool, perfect solution. 

Only annoying thing for me is running the tests locally from a dedicated environment, I get:

```
FAILED tests/test_cookie.py::test_check_conda_environment_creation_and_run_all_checks - subprocess.CalledProcessError: Command '('bash', '-l', '-c', 'conda run -n __test__ pre-commit run --all-files')' returned non-zero exit status 1.

EnvironmentLocationNotFound: Not a conda environment: /home/xxx/miniconda3/envs/econ-project-templates/envs/__test__
```

The environment is there, but in a different location: 
```
$ conda env list
# conda environments:
#
base                     /home/xxx/miniconda3
__test__                 /home/xxx/miniconda3/envs/__test__
```

Do you see a quick solution for that? Else I'd be fine with merging this as is, thanks for all your work!

timmens commented:
On my machine (macOS) it runs locally (just tried). I haven't checked on my Linux. If it is a Windows problem then I don't have a quick solution.

I can check if it runs on my Linux tomorrow. If yes I'd say we merge and open an issue.

hmgaudecker commented:
Ubuntu 22.04 -- key is to start pytest from the econ-project-templates environment.

```shell-session
$ conda --version
conda 22.9.0
```
